vertx-web/src/test/java/io/vertx/ext/web/templ/impl/CachingTemplateEngineTest.java

```diff
@@ -0,0 +1,54 @@
+package io.vertx.ext.web.templ.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author <a href="https://github.com/mystdeim">Roman Novikov</a>
+ * @since 4/1/17.
+ */
+public class CachingTemplateEngineTest {
+
+  CachingTemplateEngineImpl engine;
+
+  @Before
+  public void setUp() {
+    engine = new CachingTemplateEngineImpl(".jade");
+  }
+
+  @Test
+  public void testAdjustLocation_Empty() {
+    assertEquals("templates/index.jade", engine.adjustLocation("templates/"));
+  }
+
+  @Test
+  public void testAdjustLocation_Ext() {
+    assertEquals("templates/index.jade", engine.adjustLocation("templates/.jade"));
```

Comment:
that is a filesystem convention but not true to webservers, for example RFC-5785: https://ma.ttias.be/well-known-directory-webservers-aka-rfc-5785/ which is a common feature for auth related tasks requires dot files and dot directories.

Besides in vertx-web we are not oppionated so we should not force it to the end user. We could however add a switch to enable/disable it though. But that should be applicable to the router i guess.